### ISSUE TITLE
asterisk filter fails with systemd backend.

### ISSUE THREAD

**Issue by nimbius** [fail2ban/fail2ban#2383]
asterisk with systemd backend fails to catch anything.  this is because the output for systemd logs is slightly different than the output for text logs for asterisk.  it is also because no journalmatch is specified.

example out from systemd:
```
Mar 27 09:40:09 dunkel.dev1ce.com asterisk[326]: [Mar 27 09:40:09] NOTICE[566]: chan_sip.c:28926 handle_request_register: Registration from '"28" <sip:28@107.161.26.204>' failed for '77.247.109.78:7998' - Wrong password
```
this can be worked around by commenting out prefixreg line 21.

full conformity with systemd journal matched entries can be confirmed with this line added.
 36 journalmatch = _SYSTEMD_UNIT=asterisk.service

**Comment by sebres** [comment]
Hmm... dual date pattern... I can swear I saw already similar issues (even using systemd-journal logging).

> this can be worked around by commenting out prefixreg line 21.

I don't think so. It looks like the message (without timestamp and info) starts with `[Mar 27 09:40:09] NOTICE[566]: ...`. 
In this case commenting out the `prefregex` would hinder the proper recognition of failure with `failregex` (at least with current filter of 0.10).

Could you please provide the logged message only (or json) for this message:
```
journalctl -o cat -u asterisk | grep Wrong password
journalctl -o json -u asterisk | grep Wrong password
```

**Comment by nimbius** [comment]
Sergey,
thanks for your quick response.  The output of the cat command is listed below.  the json command returned no results.


 [Mar 27 10:06:14] NOTICE[566]: chan_sip.c:28926 handle_request_register: Registration from '"1000" <sip:1000@107.161.26.204>' failed for '77.247.109.78:7998' - Wrong password

**Comment by sebres** [comment]
Perfect, thx!

**Comment by sebres** [comment]
1. As regards the double timestamp (part `[Mar 27 10:06:14] ` in journal message), I'm pretty sure:
  - either it is a mistake of asterisk and will be fixed later (journal already has datetime);
  - or it is some misconfiguration or missing restart of services after upgrade (switch to systemd), so something like https://github.com/fail2ban/fail2ban/issues/2309#issuecomment-448939153 (although syslog, but same issue)

    Anyway in the fix I'll provide it optional in journal-case.

2. As long as I working on common solution for this and some similar issues (and new version still not released), you could use simple configuration like this:

jail.local:
```ini
[asterisk]
filter = asterisk[__prefix_line="<known/__prefix_line>(?:\[[^\]]+\]\s+)?", journalmatch="_SYSTEMD_UNIT=asterisk.service"]
```
To test it you can use that:
```
fail2ban-regex systemd-journal 'asterisk[__prefix_line="<known/__prefix_line>(?:\[[^\]]+\]\s+)?", journalmatch="_SYSTEMD_UNIT=asterisk.service"]'
```
Because journal backend uses another (more shorter) prefix-line (and I will fix this in future version too), I'll recommend to use following config as `filter.d/asterisk.local` instead of changing the `jail.local` as described above.

filter.d/asterisk.local:
```ini
[Definition]

__prefix_line_journal = \s*(?:%(__hostname)s\s+)?(?:%(_daemon)s%(__pid_re)s?:?\s+)?(?:%(__kernel_prefix)s\s+)?
__extra_timestamp = (?:\[[^\]]+\]\s+)?
__prefix_line = %(__prefix_line_journal)s%(__extra_timestamp)s

journalmatch = _SYSTEMD_UNIT=asterisk.service
```

**Comment by sebres** [comment]
fixed in #2387
